File: currency/src/types/meta.rs
use chrono::{DateTime, Utc};
use serde::Serialize;

#[derive(Debug, Clone)]
pub struct MetaFactors {
    pub hash: String,
    pub version: String,
    pub start_time: DateTime<Utc>,
}

#[derive(Debug, Serialize, Clone)]
pub struct Meta {
    pub hash: String,
    pub version: String,
    pub uptime: i64,
}

File: wallet/utils/meta.go
package utils

import (
	"os"
	"strings"
	"time"

	"github.com/autobar-dev/services/wallet/types"
)

func MetaFactorsToMeta(mf types.MetaFactors) *types.Meta {
	now := time.Now()
	uptime := int64(now.Sub(mf.StartTime).Seconds())

	return &types.Meta{
		Hash:    mf.Hash,
		Version: mf.Version,
		Uptime:  uptime,
	}
}

func GetMetaFactors() *types.MetaFactors {
	commit_sha := ""
	version := ""

	if commit_sha_bytes, err := os.ReadFile(".meta/HASH"); err == nil {
		commit_sha = strings.TrimSpace(string(commit_sha_bytes))
	}

	if version_bytes, err := os.ReadFile(".meta/VERSION"); err == nil {
		version = strings.TrimSpace(string(version_bytes))
	}

	return &types.MetaFactors{
		Hash:      commit_sha,
		Version:   version,
		StartTime: time.Now(),
	}
}
